### DISCUSSION TITLE
ISR: incorrect behavior or concept misunderstanding

### DISCUSSION THREAD

**Discussion by alyustik** [vercel/next.js#31730] [Help] [answered]
Hey guys! I've faced an issue with ISR mechanism.
I've found a scenario at which my pages are not rebuilding and I'm not sure on which side is a problem.
I have pages with slugs (e.g. articles), so my getStaticPaths() function looks like:
```
export async function getStaticPaths() {
	const articlesSlugs = await getSlugs();
	return {
		paths: articlesSlugs,
		fallback: 'blocking'
	};
}
```
and my getStaticProps function looks like:
```
export async function getStaticProps(context) {
	const {params: {slug}} = context;
	const pageData = await articlesDetailApi.get(slug);
	const errorCode = pageData?.errorCode;
	if (errorCode || isEmpty(pageData)) {
		return {
			notFound: true
		};
	}
	return {
		props: {
			store: {
				articlesDetail: {
					[slug]: articlesDetail
				},
			}
		},
		revalidate: 10
	};
}
```
The scenario which ends up with unexpected behavior is following:
1. I create an article with slug "test". ALL FINE: I check this page on site and successfully get it
2. I edit this article and change the slug to "test-new". ALL FINE: "test-new" returns the article, "test" returns 404
3. I create a new article with slug "test", which was previously occupied. ERROR: I always get 404 on this slug, ISR not triggering for this page. Only full rebuild fixing this issue

**Comment by wwwdepot** [comment, marked as answer]
Use revalidate

```jsx
if (errorCode || isEmpty(pageData)) {
    return {
        notFound: true,
        revalidate: 10
    };
}
```

**Reply by alyustik** [reply]
thanks, it helped!
